### ISSUE TITLE
Colour options / settings / preferences

### ISSUE THREAD

**Issue by wfaulk** [chrisboyle/sgtpuzzles#20]
I'd like to see options to customize colors. Light yellow on light gray can be really hard to see on the line-based games.

**Comment by chrisboyle** [comment]
I changed the Map colours in 160012ba78b89ae1c5534a1ff87c6349f53a7d1f (improving colour blind accessibility and night mode compatibility) and have two complaints so far. David B. Hecht writes:
> I am disappointed that the colors in Map are now bright primaries instead of the previous earth tones, and I can't seem to change them in settings. The red seriously bothers me for some reason.

and Linda George asks in a Play review to
> change the colours back to "camo" in the maps puzzle, or even better, let us choose our own

**Comment by fropeter** [comment]
It would be an exaggeration to call me an unhappy user, but I do have a problem with this game:
When marking possible colors for an area, i.e. using dots, some of the colors are hard to distinguish (blue/green) and yellow is hard to see on a white background. If there was a setting to choose between original and current colorscheme, (apart from my liking the originals better,) I think it would solve this particular problem.

**Comment by omgitsraven** [comment]
I would also enjoy the ability to customize all colours (i.e. even black + white) just for aesthetic reasons, if possible. Colourblind accessibility is the most important reason to make this change, but once it's in, hopefully it would be doable to not just stop at hard-coding alternate sets for the colourblind-problematic games, but having a section of the preferences to tweak all of it at will?

**Comment by chrisboyle** [comment]
Just to explicitly note it here: Simon has mentioned a plan to add colour preferences upstream, so I will wait for that to happen before adding anything on Android.

**Comment by NA0341** [comment]
Thank you for pointing that out :)
Most Applications I use have one Setting for Dark Mode¹ and one for the Theme Variant² which works pretty well.

¹ i.e. `light, dark, system`
² for dark this could be `teal, dark, black` and `default, bright, sakura, etc` for bright

---

Since this is more than one simple application, offering users the ability to create their own color palette may be a good idea.

**Suggestion**
You could add a button in settings: `create custom theme` which would create a text file containing all color values from current color scheme (which is saved using the systems save dialog).
The user can then edit the file externally and apply it with `apply custom theme`.

This would also allow sharing themes easily

**Comment by slacksystem** [comment]
I agree about customizable colors being the best solution. Being able to set each of the puzzle related colors and background is surely the most complete possible solution in terms of both accessibility and user preference, and should leave no room for anyone to complain about it if implemented.   @chrisboyle do you know whether or not that's what Simon plans to add or if he might be thinking more like a couple hard-coded options? Not sure how much he divulged of his plan

**Comment by trajano** [comment]
Yeah. Being partially color blind most of the games are not playable for me.

**Comment by NA0341** [comment]
> A user wrote in 2013 ([#73](https://github.com/chrisboyle/sgtpuzzles/issues/73)):
> 
> "Hi Chris. I'm playing Keen and have run into a problem. I'm color blind. Do you think you could change the colors to not be green and red so that I can see when I've made a mistake?"

Well, I think colorblindness is something to be taken care of by the OS. Because Android has native colorblindness settings.

@trajano
Android Settings -> Accessibility -> Visibility Enhancements -> Color Adjustment
(Exact Name and location might be different for you. I'm on a Samsung Galaxy S8. But you should be able to find a using search. Just search for color.)

Because with this you have actual system-wide color correction. Not just in one game.

**Comment by slacksystem** [comment]
Yes, the os has colorblind filters, and yes, they should be taken care of by the OS. Yet most times they are quite insufficient to help one distinguish all colors. The use of red and green, for example, in dominosa, is not enough changed by the OS to really be distinguishable for me without strain and missing some sometimes. The red on black too. But yeah, it should be taken care of by the os but it's not for a bunch of colorblind people hence the request.
